lib/facility.rb

```diff
@@ -1,14 +1,76 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, 
+              :address, 
+              :phone, 
+              :services, 
+              :registered_vehicles,
+              :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(info)
+    @name = info[:name]
+    @address = info[:address]
+    @phone = info[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def has_service?(service)
+    @services.include?(service)
+   end
+
+  def allow_service(service)
+    if has_service?(service)
+      yield
+    else
+      false
+    end
+  end
+
+  def register_vehicle(vehicle)
+    if 
+      @registered_vehicles << vehicle
+      vehicle.set_date
+      vehicle.give_plate
```

Comment:
I like that you created these two helper methods in the Vehicle class to handle the changing of the attributes.